src/openssl/mod.rs

```diff
@@ -4,82 +4,69 @@ use assert_fs::{
     fixture::{ChildPath, PathChild},
     TempDir,
 };
-use bawawa::{Command, Error, Process, Program, StandardError, StandardOutput};
-use futures::stream::Stream;
 use std::path::PathBuf;
-use tokio_codec::LinesCodec;
+use std::process::Command;
+use std::str::FromStr;
+type Error = std::io::Error;
+use crate::process::output_extensions::ProcessOutput;
+
 pub struct Openssl {
-    program: Program,
+    program: PathBuf,
 }
 
 impl Openssl {
     pub fn new() -> Result<Self, Error> {
         Ok(Openssl {
-            program: Program::new("openssl".to_owned())?,
+            program: PathBuf::from_str("openssl").unwrap(),
         })
     }
 
     pub fn version(&self) -> Result<String, Error> {
-        let mut openssl = Command::new(self.program.clone());
-        openssl.argument("version");
-        self.echo_stdout(openssl)
-    }
-
-    fn echo_stdout(&self, cmd: Command) -> Result<String, Error> {
-        let captured = Process::spawn(cmd.clone())?
-            .capture_stdout(LinesCodec::new())
-            .capture_stderr(LinesCodec::new())
-            .wait();
-        println!("{}", cmd);
-
-        let lines: Vec<String> = captured
-            .map(|r| r.unwrap_or_else(|_| "".to_owned()))
-            .collect();
-        Ok(lines.join("\n"))
+        Ok(Command::new(self.program.clone())
+            .arg("version")
+            .output()?
+            .as_lossy_string())
```

Comment:
This lossy string conversion is probably OK for tests, because any "lossy" result will not match the expectation.
But I'd rather change the extension trait to do `String::from_utf8(output).unwrap()` whenever UTF-8 or ASCII output is expected, just to be sure.

Reply:
The `as_` conversions are also wrongly named: if you allocate the string, the conversion is not zero-cost and so should be named something like `to_string`. Or `stdout_string`, `stderr_string` because you are converting some part of the output.